### PR TITLE
Feature 1795 layer search

### PR DESCRIPTION
**Pull request by yvonnesjy**
Add searching functionality to the layers panel- compatible with both "layers" view and "layerCategories" view. Update the view finder panel to share the search box component with the layers panel.

Fixes https://github.com/NCEAS/metacatui/issues/1795

NOTE: The base of this PR is feature-1773-layer-categories because the search implementation is an extension of the category work. I want to get the review started now. When the base branch is submitted, I'll create a new proper PR that references this one.

### CODE REVIEW

File: src/js/views/maps/SearchInputView.js
Comment (yvonnesjy) [inline src/js/views/maps/SearchInputView.js:1]: The implementation here isn't exactly the same as viewfinder, though the functionality should be noop.

@ianguerin lmk if you have any concern.

File: src/css/map-view.css [hunk 1]
```diff
@@ -37,6 +37,10 @@
   --map-col-yellow: #f8b406;
   --map-col-buttons-emphasis: var(--portal-secondary-color, var(--map-col-green));
   --map-col-highlight: var(--portal-primary-color, var(--map-col-blue));
+  --map-input-box-error-background: var(--portal-secondary-color-transparent);  /* #FFFCF4; */
```
Comment (ianguerin) [inline src/css/map-view.css]: I think the variable value comment is likely to become stale if the variable changes in the future

File: src/js/views/maps/LayerItemView.js [hunk 2]
```diff
@@ -486,9 +489,35 @@ define(
               '. Error details: ' + error
             );
           }
-        }
+        },
 
+        /**
+         * Searches and only dispays self if layer label matches the text. Highlights the
```
Comment (ianguerin) [inline src/js/views/maps/LayerItemView.js]: s/dispays/displays

File: src/js/views/maps/LayerListView.js [hunk 3]
```diff
@@ -119,47 +119,43 @@ define(
...
-              'There was an error rendering a LayerListView' +
-              '. Error details: ' + error
-            );
-          }
+          return this;
         },
 
-
+        /**
+         * Searches and only dispays layers that match the text.
```
Comment (ianguerin) [inline src/js/views/maps/LayerListView.js]: s/dispays/displays

File: src/js/views/maps/ViewfinderView.js
Comment (ianguerin) [inline src/js/views/maps/ViewfinderView.js:1]: Can we hold off on migrating ViewfinderView to reusable component, this is going to create some very complex merge conflicts

File: src/js/views/maps/SearchInputView.js
Comment (ianguerin) [reply src/js/views/maps/SearchInputView.js:1]: I can't seem to find any consistent guidance on how to create shared components/widgets (whatever they're called in Backbone) using Backbone JS. I asked Robyn for an example in our codebase and am waiting to hear back.

Without having to update the shared component each time that I want to tweak the functionality of my implementation code, I'm imagining a slightly less stateful View that would just make it easy to share CSS, manage the "cancel" vs "search" button visibility. Then as a user of this shared view, I can bind my own events (I need keyup, keydown, change, etc..)

File: src/css/map-view.css (hunk 1 above)
Comment (yvonnesjy) [reply src/css/map-view.css]: oh I forgot to add a proper comment- it is the actual yellow color from the mock. I'll incorporate it in the next UI update PR.

File: src/js/views/maps/ViewfinderView.js
Comment (yvonnesjy) [reply src/js/views/maps/ViewfinderView.js:1]: For sure!

**Comment by yvonnesjy** [comment]
@ianguerin Thinking through the requirements- to share CSS and cancel/search visibility, at the minimum we need a View class, sharing a template won't be sufficient. Then the customizable events will trigger from this class. Does it work if we just pass in the event handlers from parent view?

Alternatively, inheritance? https://danhough.com/blog/backbone-view-inheritance/

File: src/css/map-view.css (hunk 1 above)
Comment (yvonnesjy) [reply src/css/map-view.css]: Updated

File: src/js/views/maps/LayerListView.js (hunk 3 above)
Comment (yvonnesjy) [reply src/js/views/maps/LayerListView.js]: Done

File: src/js/views/maps/LayerItemView.js (hunk 2 above)
Comment (yvonnesjy) [reply src/js/views/maps/LayerItemView.js]: Done

File: src/js/views/maps/SearchInputView.js
Comment (yvonnesjy) [reply src/js/views/maps/SearchInputView.js:1]: Pushed a new commit for what we discussed. I left the new event handlers out for you to add when you update ViewfinderView

**Comment by yvonnesjy** [comment]
FYI the force push here^ is because the base branch got more commits- idt there's a way around it.

**Pull request closed without merging**